docs/contributing/writing-instrumentation.md

```diff
@@ -124,3 +124,27 @@ are not legal in Java 7 (and prior) bytecode.
 Because OpenTelemetry API has many common default methods (e.g. `Span.current()`),
 the `javaagent-api` artifact has a class `Java8BytecodeBridge` which provides static methods
 for accessing these default methods from advice.
+
+### Why hard code advice class names?
+
+Subclasses of `InstrumentationModule` will often implement advice classes as static inner classes.
```

Comment:
`transformers()` method belongs to the `TypeInstrumentation` interface and I think it's a bit better to focus on them in this particular gotcha. A module just brings everything together.

```suggestion
Implementations of `TypeInstrumentation` will often implement advice classes as static inner classes.
```

Reply:
It might be useful to mention that it's okay to keep the advice class in the module class or in the package.